core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -999,6 +1005,15 @@ public PlanOptimizers(
                         new PushPartialAggregationThroughExchange(plannerContext),
                         new PruneJoinColumns(),
                         new PruneJoinChildrenColumns())));
+        // This rule does not touch query plans, but only add subfields if necessary. Trigger at the near end
```

Comment:
Are there queries that actually fail to get pushdown if we don't have this rule here? Maybe it's not needed?